src/mentorpal/classifiers/arch/lstm_v2/training/__init__.py

```diff
@@ -68,144 +91,117 @@ def __load_training_data(self, path):
             questions=corpus.iloc[i]['question'].split('\n')
             questions=[_f for _f in questions if _f]
             current_question=questions[0]
-            processed_question=preprocessor.transform(current_question) # tokenize the question
             # answer
             answer=corpus.iloc[i]['text']
             answer_id=corpus.iloc[i]['ID']
             answer=answer.replace('\u00a0',' ')
-            #Generate text features
-            negation_feature = features.any_negation(current_question)
-            wordcount_feature = features.log_wordcount(current_question)
-            negmod_feature = features.negation_mod(current_question)
-            whatques_feature = features.what_question(current_question)
-            howques_feature = features.how_question(current_question)
-            whyques_feature = features.why_question(current_question)
-            whenques_feature = features.when_question(current_question)
-            whereques_feature = features.where_question(current_question)
+            # univariate features
+            univariate_features_question = textfeaturegen.univariate_features(current_question)          
             #add question to dataset
-            # train_data.append([current_question,processed_question,topics,answer_id,answer])
-            train_data.append([current_question,processed_question,topics,answer_id,answer,negation_feature,wordcount_feature,negmod_feature,whatques_feature,howques_feature,whyques_feature,whenques_feature,whereques_feature])
-            
+            processed_question=preprocessor.transform(current_question) # tokenize the question
+            # ERROR BELOW: adding None to training_data because <list>.append([]) returns None
+            train_data.append([current_question,processed_question,topics,answer_id,answer].extend(univariate_features_question))
```

Comment:
`train_data` is going to be a list full of `None` because `<list>.extend([])` returns `None`

Reply:
oh, my bad! How could I have looked over this one..?!

My basic idea was to extend and not append to the existing training row. This is because `univariate_features_question` itself returns a list.

Moving `.extend()` to another line.